File: api/src/endpoint/question/update_question.rs
use serde::{Deserialize, Serialize};
use api_derive::EndpointRequest;

use crate::{endpoint::Endpoint, resource::question::QuestionResource};

pub struct UpdateQuestion;

impl Endpoint for UpdateQuestion {
    const PATH: &'static str = "/repository/{owner}/{repo}/question/{number}";
    const METHOD: http::Method = http::Method::PATCH;

    type Request = UpdateQuestionRequest;
    type Response = UpdateQuestionResponse;
}

#[derive(EndpointRequest, Debug, Serialize, Deserialize)]
pub struct UpdateQuestionRequest {
    pub title: String,
    pub body: String,
}

pub type UpdateQuestionResponse = QuestionResource;

File: gitdot-api-ts/src/endpoint/question/update-question.ts
import { z } from "zod";

import { QuestionResource } from "../../resource";

export const UpdateQuestionRequest = z.object({
  title: z.string(),
  body: z.string(),
});
export type UpdateQuestionRequest = z.infer<typeof UpdateQuestionRequest>;

export const UpdateQuestion = {
  path: "/repository/{owner}/{repo}/question/{number}",
  method: "PATCH",
  request: UpdateQuestionRequest,
  response: QuestionResource,
} as const;
export type UpdateQuestion = typeof UpdateQuestion;
